ext/test/http/curl_http_test.cc

```diff
@@ -200,9 +202,10 @@ TEST_F(BasicCurlHttpTests, SendGetRequest)
 TEST_F(BasicCurlHttpTests, SendPostRequest)
 {
   received_requests_.clear();
-  curl::SessionManager session_manager;
+  auto session_manager = http_client::HttpClientFactory::Create();
+  EXPECT_TRUE(session_manager != nullptr);
 
-  auto session = session_manager.CreateSession("127.0.0.1", HTTP_PORT);
+  auto session = session_manager->CreateSession("127.0.0.1", HTTP_PORT);
   auto request = session->CreateRequest();
```

Comment:
Shouldn't `request` be not even tied to particular implementation of a client? In a model where `request` is generic, you can create it, specify parameters, headers, destination URL host / port. Then fire it via client, which then will create  (or reuse) the connection, etc.. Basically Session Manager then is becoming totally unneeded. I'm trying to understand why we require it as a separate class. This could be detail hidden in the client itself, which should auto-manage the connection, SSL handshake, maintain the handshake for persistent connections (without giving much info to user about it), and should be able to handle incoming requests. The requests objects should not need to be client-specific: any HTTP client impl may be able to handle a generic request object that contains the destination, body, headers, etc.

Reply:
Yes that is also an option - We can get rid of `Session` object and have `Request` managed directly with `SessionManager`.  But I don't see that scope of this PR ? Although we already reviewed and approved http interface PR, I would be fine revisiting it again separately.